week-2-wide-perspective/w02e05-simple-server-side/src/server.ts

```diff
@@ -0,0 +1,15 @@
+import dotenv from "dotenv";
+import express from "express";
```

Comment:
tutaj powinien być dorzucony jeszcze import:

```
import 'express-async-errors'
```

Chodzi o błąd `express'a` który pomimo tego że używać może `async  / await` to nie jest na to "Gotowy" z perspektywy czasu - kiedy ostatnio został wydany.

Chodzi tutaj o podobny błąd jaki mieliśmy w zadaniu z żarówką `w01e04` - jak tam robiliśmy sobie `errorHandler` na szczycie - to łapał wszystko.... czy aby na pewno wszystko ? Noo.... nie wszystko bo był problem z błędami rzucanymi w `setTimeout` i trzeba było napisać specjalnie obsługę upływu czasu jako `async  / await`  i zmienić sposób rzucania błędem.

TUTAJ: sytuacja jest totalnie analogiczna.

używając `async / await` w kontrolerach nie masz pewności, że jeśli coś rzuci błędem, to to o czym Ci piszę poniżej `globalErrorHanlder` - złapie błąd.
Właśnie on go nie wyłapie, chyba że zastosujesz `express-async-errors` - to pewnego rodzaju "hack" - ale działa.